src/blech32.rs

```diff
@@ -144,23 +179,24 @@ pub fn decode(s: &str) -> Result<(&str, Vec<u5>), Error> {
     }
 
     // Ensure checksum
-    if !verify_checksum(&hrp_bytes, &data) {
-        return Err(Error::InvalidChecksum);
-    }
+    match verify_checksum(&raw_hrp.as_bytes(), &data) {
+        Some(variant) => {
+            // Remove checksum from data payload
+            let dbl: usize = data.len();
+            data.truncate(dbl - 12);
```

Comment:
You lost the `// ELEMENTS: 6->12` comment here though. Is that still relevant or was it removed on purpose?

Reply:
It is still relevant, I removed it by accident